File: cpp/apcs/c461/c461.cpp
#include<iostream>

using namespace std;
int main(){
    short d,e;
    bool a,b,c,f=false;
    cin>>d>>e>>c;
    a=!!d;
    b=!!e;
    if((a&&b)==c){
        cout<<"AND"<<endl;
        f=true;
    }
    if((a||b)==c){
        cout<<"OR"<<endl;
        f=true;
    }
    if((a^b)==c){
        cout<<"XOR"<<endl;
        f=true;
    }
    if(f==false){
        cout<<"IMPOSSIBLE";
    }
}

File: python/apcs/第一題/c461.py
x=False
a,b,c=map(bool,map(int,input().split()))
if (a and b)==c:
    print('AND')
    x=True
if (a or b)==c:
    print('OR')
    x=True
if (a ^ b)==c:
    print('XOR')
    x=True
if x==False:
    print('IMPOSSIBLE')